src/CommandHandler.py

```diff
@@ -129,7 +129,28 @@ def CreateLabHandler(self, cmd: [str]):
         print('Create lab:', cmd)
 
     def AssignCourseHandler(self, cmd: [str]):
-        print('Assign course:', cmd)
+        if self.currentUser is None or not self.currentUser.RoleIn(Role.Administrator, Role.Supervisor):
+            print('Must be logged in as an Administrator or a Supervisor')
+            return
+        if len(cmd) != 2:
+            print('Invalid number of arguments')
+        c = Course(self.db)
+        c.course_name = cmd[0]
+        if not c.Exists():
+            print('This course does not exist.')
+            return
+        c.GetDetail()
+        c.instructor_email = cmd[1]
+
+        acc = Account(self.db)
+        acc.act_email = cmd[1]
+        acc.GetDetail()
+        if acc.role_id is None or not acc.RoleIn(Role.Instructor):
```

Comment:
We don't need to check if role_id is None here, since role_id is 'not null'

Reply:
If we do want extra redundancy for safety, we could move the null check into Account.RoleIn.  In that case, it would always return false if role_id is None.